First version of json description

stity opened the pull request:
I wrote a nodejs script to generate the JSON from the CSV.
Run it to see : 
- Number of data sources
- Number of structures
- Structures without data source
- VTK files not used as data source

mhalle commented:
Thanks! Couple of ideas.
- For the datasource, I would use a relative URL without a scheme for the source files.  I.E:
  "source": "models/Model_19_..."
- create a new set of structures that reference the labelmap NRRD file in slicer/volumes/labels for the volumetric representation.  There should be one structure per label number. They should reference a single datasource (the NRRD file) using a sourceSelector with a dataKey. Note we don't have a way of associating the geometric and volumetric structures together yet. I've got some ideas.
- The source selectors for the vtk models don't need dataKeys because the geometry files only include a single structure.  
- you can omit the "notes" field.  The only reason it exists is to provide a unique ID for some structures that RADLEX didn't split up.  I consider these "notes" obsolete.
- please put the script in its own directory (tsvToJson) under a toplevel utility "utils".  That way if you need any npm modules in the future, they'll be contained.
- Yes, it looks like the colors are screwed up, thanks for noticing.  I believe the other tsv file in the labelinfo directory has the correct colors, matching the color table (please check).  If you could reconcile them, that would be great. 
- I'll check on the mysterious extra vtk files.
- You use sequential ID generators.  You might experiment with UUID4 generators to create the IDs.  You can use any of the NPM modules for uuid (there are several).

Commit:
move module to its own directory

Commit:
change directory + nrrd parser + color matching + code restructuring + small improvement

stity commented:
I have made all the suggested changes.
And added a nrrd parser (had to write my own as the npm one does not handle gzip data because it is not required for nrrd parser to handle gzip)
I added a function to check if the colors used in hncma tsv and the color table match for a given value.
Thanks for reviewing the code, to be sure that I understood the file format correctly.

stity commented:
The log generated by node (run it to see the colors) : 

``` console
320 vtk files found
317 data source(s) created in vtk conversion
Warning useless vtk files :  [
    "Model_215_right_optic_nerve.vtk",
    "Model_300_left-I_II-v.vtk",
    "Model_77_left_optic_nerve.vtk"
]
335 structure(s) created in vtk conversion
Warning structures without vtk files :  [
    "left cerebellar cortex",
    "left thalamus",
    "left middle cerebellar peduncle",
    "unsegmented region of left hypothalamus",
    "choroid plexus of left cerebral hemisphere",
    "choroid plexus of left cerebral hemisphere",
    "right cerebellar cortex",
    "right thalamus",
    "unsegmented region of right hypothalamus",
    "choroid plexus of right cerebral hemisphere",
    "left VIIAf v",
    "left VIIAt v",
    "left X m",
    "right VIIAf v",
    "right VIIAt v",
    "right X m",
    "right angular and supramarginal gyri (obsolete label)"
]
hncma-atlas.nrrd has been parsed with success
skin.nrrd has been parsed with success
2 data sources created from nrrd files
336 structure(s) created from nrrd files
Warning :   no match found in the nrrd files for these structure :  [
    "left cerebellar cortex",
    "left middle cerebellar peduncle",
    "left hypothalamus",
    "choroid plexus of left cerebral hemisphere",
    "left superior cerebellar peduncle",
    "right cerebellar cortex",
    "right hypothalamus",
    "choroid plexus of right cerebral hemisphere",
    "right angular and supramarginal gyri (obsolete label)",
    "left calcarine sulcus",
    "right calcarine sulcus",
    "left central sulcus",
    "right central sulcus",
    "left cingulate sulcus",
    "right cingulate sulcus",
    "left collateral sulcus",
    "right collateral sulcus",
    "left inferior frontal sulcus",
    "right inferior frontal sulcus",
    "left intraparietal sulcus",
    "right intraparietal sulcus",
    "left parieto-occipital sulcus",
    "right parieto-occipital sulcus",
    "left postcentral sulcus",
    "right postcentral sulcus",
    "left precentral sulcus",
    "right precentral sulcus",
    "left superior frontal sulcus",
    "right superior frontal sulcus",
    "left superior temporal sulcus",
    "right superior temporal sulcus"
]
The JSON file was saved!
done in 1487ms
```

Commit:
add : check if there is a label in the nrrd file that is not is the tsv

stity commented:
With the new version I get that the value 10 is in the nrrd files but not in the tsv

Commit:
add : xtk visualisation of the atlas using a modified version of xtk to be able to load binary vtk files
For more info about this modified version see @dblezek work at https://github.com/xtk/X/pull/164

Commit:
add : small nodejs server to be able to execute XHR on localhost

Commit:
add : hierarchy extraction using the python script and integration of the hierarchy with groups in the json file

Commit:
add : mark the root group with an annotation to be able to easily built the tree later

Commit:
add : hierarchy tree in the xtk demo + bower as component manager for the page

Commit:
fix : style of the tree list

Commit:
fix : missing semi colon

Commit:
fix : forgot to add the nrrd files to the json file

Commit:
add : working demo of the atlas using three.js (missing the tree)

Commit:
add : begining of threejs application with the tree hierarchy (not working)

Commit:
add : fully working three.js demo

Commit:
fix : initial aspect of the renderer

Commit:
fix : add $apply to commit changes to the progress bar on signal

Commit:
enh : cleaning

Commit:
fix : rotation and indentation

Commit:
enh : switch root attribute of groups from annotation object to group object

Commit:
del : delete three directory to replace it by a submodule

Commit:
add : viewer submodule

Commit:
add : config file to submodule

stity committed:
enh : update submodule

Signed-off-by: Valentin Demeusy <valentin.demeusy@gmail.com>

stity committed:
Create gh-pages branch via GitHub

stity committed:
Merge branch 'master' into gh-pages

stity committed:
enh : update

Signed-off-by: Valentin Demeusy <valentin.demeusy@gmail.com>

stity committed:
fix : switch json extraction from tsv to mrml

stity committed:
Merge branch 'master' into gh-pages

stity committed:
fix : put atlasStructure.json in utils to be able to commit it to gh-pages

stity committed:
enh : update

Signed-off-by: Valentin Demeusy <valentin.demeusy@gmail.com>

stity committed:
enh : update submodule

stity committed:
enh : update

Signed-off-by: Valentin Demeusy <valentin.demeusy@gmail.com>

stity committed:
Merge branch 'master' into gh-pages

Conflicts:
	utils/configAtlasViewer.js
	utils/configMrmlToJson.js

Signed-off-by: Valentin Demeusy <valentin.demeusy@gmail.com>

stity committed:
fix : change atlasStructure location

stity committed:
enh : update

stity committed:
Merge branch 'master' into gh-pages

Conflicts:
	utils/atlasStructure.json

Signed-off-by: Valentin Demeusy <valentin.demeusy@gmail.com>

stity committed:
enh : update

stity committed:
Merge branch 'master' of http://github.com/stity/spl-brain-atlas into gh-pages

Conflicts:
	utils/atlasStructure.json

Signed-off-by: Valentin Demeusy <valentin.demeusy@gmail.com>

stity committed:
enh : update atlas viewer

Signed-off-by: Valentin Demeusy <valentin.demeusy@gmail.com>

stity committed:
enh : update

Signed-off-by: Valentin Demeusy <valentin.demeusy@gmail.com>

stity committed:
enh : update

Signed-off-by: Valentin Demeusy <valentin.demeusy@gmail.com>

stity committed:
enh : update

Signed-off-by: Valentin Demeusy <valentin.demeusy@gmail.com>

stity committed:
enh : update

Signed-off-by: Valentin Demeusy <valentin.demeusy@gmail.com>

stity committed:
enh : update

Signed-off-by: Valentin Demeusy <valentin.demeusy@gmail.com>

stity committed:
enh : update

Signed-off-by: Valentin Demeusy <valentin.demeusy@gmail.com>

stity committed:
enh : update and run new mrmlToJson script

Signed-off-by: Valentin Demeusy <valentin.demeusy@gmail.com>

stity committed:
Merge branch 'master' of https://github.com/mhalle/spl-brain-atlas into gh-pages

stity committed:
add : label map in the viewer (using resampled images)

stity committed:
enh : update

stity committed:
enh : update

Signed-off-by: Valentin Demeusy <valentin.demeusy@gmail.com>

stity committed:
enh : update

Signed-off-by: Valentin Demeusy <valentin.demeusy@gmail.com>

stity committed:
enh : update

Signed-off-by: Valentin Demeusy <valentin.demeusy@gmail.com>

stity committed:
enh : update

Signed-off-by: Valentin Demeusy <valentin.demeusy@gmail.com>

stity committed:
enh : update

Signed-off-by: Valentin Demeusy <valentin.demeusy@gmail.com>

stity committed:
enh : update

Signed-off-by: Valentin Demeusy <valentin.demeusy@gmail.com>

stity committed:
enh : update readme

Signed-off-by: Valentin Demeusy <valentin.demeusy@gmail.com>

stity committed:
del : gh-pages index page

Signed-off-by: Valentin Demeusy <valentin.demeusy@gmail.com>
